### PR TITLE
Allow Single Top Half Slab to be Snowable

### PR DESCRIPTION
**Pull request by Xenoxis**
It allow a single slab (the top-half part) to be snowable.

Currently i haven't understand how bindings works, so i'm not quite sure about Lua parts in src\Bindings\DepreacatedBindings.h and the function `tolua_get_AllToLua_g_BlockIsSnowable` if something more proper about it can be done or not.

### CODE REVIEW

**Comment by bearbin** [comment]
Your commit history is all messed up, creating conflicts with work that's been done on master. 

You need to rebase the relevant commits for this PR on top of the current upstream master.

**Comment by Xenoxis** [comment]
@bearbin I don't know how to do it, can you explain to me ?

**Comment by bearbin** [comment]
It's not necessary now, you appear to have resolved it with a merge instead. But basically when you are creating a new branch for a pull-request, you need to split off the cuberite/master branch and not the master branch of your fork.

File: src/BlockInfo.h [hunk 1]
```diff
@@ -37,6 +37,38 @@ class cBlockInfo
 	{
 		return ((Get(a_Type).m_IsSkylightDispersant) || (Get(a_Type).m_SpreadLightFalloff > 1));
 	}
+	
+	inline static bool IsSingleSlab               (BLOCKTYPE a_Type)
```
Comment (bearbin) [inline src/BlockInfo.h]: No need to indent these parameters, since it isn't a one-liner.

**Comment by Xenoxis** [comment]
@bearbin I don't have permission to create a branch nor uploading on any branch of cuberite :/

**Comment by bearbin** [comment]
> @bearbin I don't have permission to create a branch nor uploading on any branch of cuberite :/

You don't need to have write permission to fetch the main repository as another remote and create a new branch off it:

```
git remote add upstream https://github.com/cuberite/cuberite.git
git fetch upstream
git checkout -b mynewfeature upstream/master
```

**Comment by bearbin** [comment]
This needs conflict resolution and the style fixing, it would be good to get this merged soon and not just have it sitting around blocked.

**Comment by Xenoxis** [comment]
I close this pull request, i will do this more proper

**Pull request closed without merging**